rclpy/rclpy/action/client.py

```diff
@@ -403,24 +455,33 @@ def send_goal(self, goal, **kwargs):
 
         event = threading.Event()
 
-        def unblock(future):
+        def unblock(future: Future[Any]) -> None:
             nonlocal event
             event.set()
 
         send_goal_future = self.send_goal_async(goal, **kwargs)
         send_goal_future.add_done_callback(unblock)
 
         event.wait()
-        if send_goal_future.exception() is not None:
-            raise send_goal_future.exception()
+        exception = send_goal_future.exception()
+        if exception is not None:
+            raise exception
 
         goal_handle = send_goal_future.result()
 
+        if not isinstance(goal_handle, ClientGoalHandle):
+            raise TypeError(
+                'Expected type ClientGoalHandle but received {}'.format(type(goal_handle)))
```

Comment:
instead of checking and returning None, this only checks the type and raise the exception if necessary, related to https://github.com/ros2/rclpy/pull/1361#discussion_r1770615712.

to be honest, i really do not see why previous change `return None` causes the segmentation fault with RHEL unless we have full stack trace what went wrong.

Reply:
Yeah I don't disagree. But It seemed the most likely cause. Looking at the trace there is no obvious cause.


The full stack trace Attached.
```
Fatal Python error: Segmentation fault
02:21:53 
02:21:53 Current thread 0x00007f49407c8740 (most recent call first):
02:21:53   File "/usr/lib/python3.9/site-packages/flake8_quotes/__init__.py", line 294 in __init__
02:21:53   File "/usr/lib/python3.9/site-packages/flake8_quotes/__init__.py", line 189 in <listcomp>
02:21:53   File "/usr/lib/python3.9/site-packages/flake8_quotes/__init__.py", line 189 in get_noqa_lines
02:21:53   File "/usr/lib/python3.9/site-packages/flake8_quotes/__init__.py", line 181 in run
02:21:53   File "/usr/lib/python3.9/site-packages/flake8/checker.py", line 500 in run_ast_checks
02:21:53   File "/usr/lib/python3.9/site-packages/flake8/checker.py", line 597 in run_checks
02:21:53   File "/usr/lib/python3.9/site-packages/flake8/checker.py", line 687 in _run_checks
02:21:53   File "/usr/lib64/python3.9/multiprocessing/pool.py", line 48 in mapstar
02:21:53   File "/usr/lib64/python3.9/multiprocessing/pool.py", line 125 in worker
02:21:53   File "/usr/lib64/python3.9/multiprocessing/process.py", line 108 in run
02:21:53   File "/usr/lib64/python3.9/multiprocessing/process.py", line 315 in _bootstrap
02:21:53   File "/usr/lib64/python3.9/multiprocessing/popen_fork.py", line 71 in _launch
02:21:53   File "/usr/lib64/python3.9/multiprocessing/popen_fork.py", line 19 in __init__
02:21:53   File "/usr/lib64/python3.9/multiprocessing/context.py", line 277 in _Popen
02:21:53   File "/usr/lib64/python3.9/multiprocessing/process.py", line 121 in start
02:21:53   File "/usr/lib64/python3.9/multiprocessing/pool.py", line 326 in _repopulate_pool_static
02:21:53   File "/usr/lib64/python3.9/multiprocessing/pool.py", line 303 in _repopulate_pool
02:21:53   File "/usr/lib64/python3.9/multiprocessing/pool.py", line 212 in __init__
02:21:53   File "/usr/lib64/python3.9/multiprocessing/context.py", line 119 in Pool
02:21:53   File "/usr/lib/python3.9/site-packages/flake8/checker.py", line 664 in _try_initialize_processpool
02:21:53   File "/usr/lib/python3.9/site-packages/flake8/checker.py", line 260 in run_parallel
02:21:53   File "/usr/lib/python3.9/site-packages/flake8/checker.py", line 309 in run
02:21:53   File "/usr/lib/python3.9/site-packages/flake8/main/application.py", line 271 in run_checks
02:21:53   File "/usr/lib/python3.9/site-packages/flake8/api/legacy.py", line 109 in check_files
02:21:53   File "/home/jenkins-agent/workspace/nightly_linux-rhel_release/ws/install/ament_flake8/lib/python3.9/site-packages/ament_flake8/main.py", line 281 in generate_flake8_report
02:21:53   File "/home/jenkins-agent/workspace/nightly_linux-rhel_release/ws/install/ament_flake8/lib/python3.9/site-packages/ament_flake8/main.py", line 88 in main_with_errors
02:21:53   File "/home/jenkins-agent/workspace/nightly_linux-rhel_release/ws/src/ros2/ros2cli/ros2cli/test/test_flake8.py", line 22 in test_flake8
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/python.py", line 183 in pytest_pyfunc_call
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/callers.py", line 187 in _multicall
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 84 in <lambda>
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 93 in _hookexec
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/hooks.py", line 286 in __call__
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/python.py", line 1641 in runtest
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 162 in pytest_runtest_call
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/callers.py", line 187 in _multicall
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 84 in <lambda>
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 93 in _hookexec
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/hooks.py", line 286 in __call__
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 255 in <lambda>
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 311 in from_call
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 254 in call_runtest_hook
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 215 in call_and_report
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 126 in runtestprotocol
02:21:53   File "/builddir/build/BUILDROOT/python-pytest-rerunfailures-9.1.1-3.el9.noarch/usr/lib/python3.9/site-packages/pytest_rerunfailures.py", line 240 in pytest_runtest_protocol
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/callers.py", line 187 in _multicall
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 84 in <lambda>
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 93 in _hookexec
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/hooks.py", line 286 in __call__
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/main.py", line 348 in pytest_runtestloop
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/callers.py", line 187 in _multicall
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 84 in <lambda>
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 93 in _hookexec
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/hooks.py", line 286 in __call__
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/main.py", line 323 in _main
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/main.py", line 269 in wrap_session
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/main.py", line 316 in pytest_cmdline_main
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/callers.py", line 187 in _multicall
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 84 in <lambda>
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 93 in _hookexec
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/hooks.py", line 286 in __call__
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/config/__init__.py", line 162 in main
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/config/__init__.py", line 185 in console_main
02:21:53   File "/usr/lib/python3.9/site-packages/pytest/__main__.py", line 5 in <module>
02:21:53   File "/usr/lib64/python3.9/runpy.py", line 87 in _run_code
02:21:53   File "/usr/lib64/python3.9/runpy.py", line 197 in _run_module_as_main
02:21:53 Fatal Python error: Segmentation fault
02:21:53 
02:21:53 Current thread 0x00007f49407c8740 (most recent call first):
02:21:53   File "<string>", line 1 in <lambda>
02:21:53   File "/usr/lib64/python3.9/tokenize.py", line 590 in _tokenize
02:21:53   File "/usr/lib/python3.9/site-packages/flake8_quotes/__init__.py", line 189 in <listcomp>
02:21:53   File "/usr/lib/python3.9/site-packages/flake8_quotes/__init__.py", line 189 in get_noqa_lines
02:21:53   File "/usr/lib/python3.9/site-packages/flake8_quotes/__init__.py", line 181 in run
02:21:53   File "/usr/lib/python3.9/site-packages/flake8/checker.py", line 500 in run_ast_checks
02:21:53   File "/usr/lib/python3.9/site-packages/flake8/checker.py", line 597 in run_checks
02:21:53   File "/usr/lib/python3.9/site-packages/flake8/checker.py", line 687 in _run_checks
02:21:53   File "/usr/lib64/python3.9/multiprocessing/pool.py", line 48 in mapstar
02:21:53   File "/usr/lib64/python3.9/multiprocessing/pool.py", line 125 in worker
02:21:53   File "/usr/lib64/python3.9/multiprocessing/process.py", line 108 in run
02:21:53   File "/usr/lib64/python3.9/multiprocessing/process.py", line 315 in _bootstrap
02:21:53   File "/usr/lib64/python3.9/multiprocessing/popen_fork.py", line 71 in _launch
02:21:53   File "/usr/lib64/python3.9/multiprocessing/popen_fork.py", line 19 in __init__
02:21:53   File "/usr/lib64/python3.9/multiprocessing/context.py", line 277 in _Popen
02:21:53   File "/usr/lib64/python3.9/multiprocessing/process.py", line 121 in start
02:21:53   File "/usr/lib64/python3.9/multiprocessing/pool.py", line 326 in _repopulate_pool_static
02:21:53   File "/usr/lib64/python3.9/multiprocessing/pool.py", line 303 in _repopulate_pool
02:21:53   File "/usr/lib64/python3.9/multiprocessing/pool.py", line 212 in __init__
02:21:53   File "/usr/lib64/python3.9/multiprocessing/context.py", line 119 in Pool
02:21:53   File "/usr/lib/python3.9/site-packages/flake8/checker.py", line 664 in _try_initialize_processpool
02:21:53   File "/usr/lib/python3.9/site-packages/flake8/checker.py", line 260 in run_parallel
02:21:53   File "/usr/lib/python3.9/site-packages/flake8/checker.py", line 309 in run
02:21:53   File "/usr/lib/python3.9/site-packages/flake8/main/application.py", line 271 in run_checks
02:21:53   File "/usr/lib/python3.9/site-packages/flake8/api/legacy.py", line 109 in check_files
02:21:53   File "/home/jenkins-agent/workspace/nightly_linux-rhel_release/ws/install/ament_flake8/lib/python3.9/site-packages/ament_flake8/main.py", line 281 in generate_flake8_report
02:21:53   File "/home/jenkins-agent/workspace/nightly_linux-rhel_release/ws/install/ament_flake8/lib/python3.9/site-packages/ament_flake8/main.py", line 88 in main_with_errors
02:21:53   File "/home/jenkins-agent/workspace/nightly_linux-rhel_release/ws/src/ros2/ros2cli/ros2cli/test/test_flake8.py", line 22 in test_flake8
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/python.py", line 183 in pytest_pyfunc_call
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/callers.py", line 187 in _multicall
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 84 in <lambda>
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 93 in _hookexec
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/hooks.py", line 286 in __call__
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/python.py", line 1641 in runtest
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 162 in pytest_runtest_call
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/callers.py", line 187 in _multicall
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 84 in <lambda>
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 93 in _hookexec
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/hooks.py", line 286 in __call__
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 255 in <lambda>
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 311 in from_call
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 254 in call_runtest_hook
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 215 in call_and_report
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 126 in runtestprotocol
02:21:53   File "/builddir/build/BUILDROOT/python-pytest-rerunfailures-9.1.1-3.el9.noarch/usr/lib/python3.9/site-packages/pytest_rerunfailures.py", line 240 in pytest_runtest_protocol
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/callers.py", line 187 in _multicall
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 84 in <lambda>
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 93 in _hookexec
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/hooks.py", line 286 in __call__
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/main.py", line 348 in pytest_runtestloop
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/callers.py", line 187 in _multicall
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 84 in <lambda>
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 93 in _hookexec
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/hooks.py", line 286 in __call__
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/main.py", line 323 in _main
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/main.py", line 269 in wrap_session
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/main.py", line 316 in pytest_cmdline_main
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/callers.py", line 187 in _multicall
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 84 in <lambda>
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 93 in _hookexec
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/hooks.py", line 286 in __call__
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/config/__init__.py", line 162 in main
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/config/__init__.py", line 185 in console_main
02:21:53   File "/usr/lib/python3.9/site-packages/pytest/__main__.py", line 5 in <module>
02:21:53   File "/usr/lib64/python3.9/runpy.py", line 87 in _run_code
02:21:53   File "/usr/lib64/python3.9/runpy.py", line 197 in _run_module_as_main
02:21:53 
02:21:53 +++++++++++++++++++++++++++++++++++ Timeout ++++++++++++++++++++++++++++++++++++
02:21:53 
02:21:53 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ Captured stderr ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
02:21:53 option --max-complexity: please update from optparse string `type=` to argparse callable `type=` -- this will be an error in the future
02:21:53 option --application-import-names: please update from optparse string `type=` to argparse callable `type=` -- this will be an error in the future
02:21:53 option --application-package-names: please update from optparse string `type=` to argparse callable `type=` -- this will be an error in the future
02:21:53 option --import-order-style: please update from optparse string `type=` to argparse callable `type=` -- this will be an error in the future
02:21:53 option --quotes: please update from optparse string `type=` to argparse callable `type=` -- this will be an error in the future
02:21:53 option --inline-quotes: please update from optparse string `type=` to argparse callable `type=` -- this will be an error in the future
02:21:53 option --multiline-quotes: please update from optparse string `type=` to argparse callable `type=` -- this will be an error in the future
02:21:53 option --docstring-quotes: please update from optparse string `type=` to argparse callable `type=` -- this will be an error in the future
02:21:53 
02:21:53 ~~~~~~~~~~~~~~~~~~~~~ Stack of Thread-28 (139950930503232) ~~~~~~~~~~~~~~~~~~~~~
02:21:53   File "/usr/lib64/python3.9/threading.py", line 937, in _bootstrap
02:21:53     self._bootstrap_inner()
02:21:53   File "/usr/lib64/python3.9/threading.py", line 980, in _bootstrap_inner
02:21:53     self.run()
02:21:53   File "/usr/lib64/python3.9/threading.py", line 917, in run
02:21:53     self._target(*self._args, **self._kwargs)
02:21:53   File "/usr/lib64/python3.9/multiprocessing/pool.py", line 576, in _handle_results
02:21:53     task = get()
02:21:53   File "/usr/lib64/python3.9/multiprocessing/connection.py", line 254, in recv
02:21:53     buf = self._recv_bytes()
02:21:53   File "/usr/lib64/python3.9/multiprocessing/connection.py", line 418, in _recv_bytes
02:21:53     buf = self._recv(4)
02:21:53   File "/usr/lib64/python3.9/multiprocessing/connection.py", line 383, in _recv
02:21:53     chunk = read(handle, remaining)
02:21:53 
02:21:53 ~~~~~~~~~~~~~~~~~~~~~ Stack of Thread-27 (139950947288640) ~~~~~~~~~~~~~~~~~~~~~
02:21:53   File "/usr/lib64/python3.9/threading.py", line 937, in _bootstrap
02:21:53     self._bootstrap_inner()
02:21:53   File "/usr/lib64/python3.9/threading.py", line 980, in _bootstrap_inner
02:21:53     self.run()
02:21:53   File "/usr/lib64/python3.9/threading.py", line 917, in run
02:21:53     self._target(*self._args, **self._kwargs)
02:21:53   File "/usr/lib64/python3.9/multiprocessing/pool.py", line 528, in _handle_tasks
02:21:53     for taskseq, set_length in iter(taskqueue.get, None):
02:21:53 
02:21:53 ~~~~~~~~~~~~~~~~~~~~~ Stack of Thread-26 (139951743956544) ~~~~~~~~~~~~~~~~~~~~~
02:21:53   File "/usr/lib64/python3.9/threading.py", line 937, in _bootstrap
02:21:53     self._bootstrap_inner()
02:21:53   File "/usr/lib64/python3.9/threading.py", line 980, in _bootstrap_inner
02:21:53     self.run()
02:21:53   File "/usr/lib64/python3.9/threading.py", line 917, in run
02:21:53     self._target(*self._args, **self._kwargs)
02:21:53   File "/usr/lib64/python3.9/multiprocessing/pool.py", line 519, in _handle_workers
02:21:53     cls._wait_for_updates(current_sentinels, change_notifier)
02:21:53   File "/usr/lib64/python3.9/multiprocessing/pool.py", line 499, in _wait_for_updates
02:21:53     wait(sentinels, timeout=timeout)
02:21:53   File "/usr/lib64/python3.9/multiprocessing/connection.py", line 935, in wait
02:21:53     ready = selector.select(timeout)
02:21:53   File "/usr/lib64/python3.9/selectors.py", line 416, in select
02:21:53     fd_event_list = self._selector.poll(timeout)
02:21:53 
02:21:53 ~~~~~~~~~~~~~~~~~~~~ Stack of MainThread (139952591243072) ~~~~~~~~~~~~~~~~~~~~~
02:21:53   File "/usr/lib64/python3.9/runpy.py", line 197, in _run_module_as_main
02:21:53     return _run_code(code, main_globals, None,
02:21:53   File "/usr/lib64/python3.9/runpy.py", line 87, in _run_code
02:21:53     exec(code, run_globals)
02:21:53   File "/usr/lib/python3.9/site-packages/pytest/__main__.py", line 5, in <module>
02:21:53     raise SystemExit(pytest.console_main())
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/config/__init__.py", line 185, in console_main
02:21:53     code = main()
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/config/__init__.py", line 162, in main
02:21:53     ret: Union[ExitCode, int] = config.hook.pytest_cmdline_main(
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/hooks.py", line 286, in __call__
02:21:53     return self._hookexec(self, self.get_hookimpls(), kwargs)
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 93, in _hookexec
02:21:53     return self._inner_hookexec(hook, methods, kwargs)
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 84, in <lambda>
02:21:53     self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/callers.py", line 187, in _multicall
02:21:53     res = hook_impl.function(*args)
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/main.py", line 316, in pytest_cmdline_main
02:21:53     return wrap_session(config, _main)
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/main.py", line 269, in wrap_session
02:21:53     session.exitstatus = doit(config, session) or 0
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/main.py", line 323, in _main
02:21:53     config.hook.pytest_runtestloop(session=session)
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/hooks.py", line 286, in __call__
02:21:53     return self._hookexec(self, self.get_hookimpls(), kwargs)
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 93, in _hookexec
02:21:53     return self._inner_hookexec(hook, methods, kwargs)
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 84, in <lambda>
02:21:53     self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/callers.py", line 187, in _multicall
02:21:53     res = hook_impl.function(*args)
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/main.py", line 348, in pytest_runtestloop
02:21:53     item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/hooks.py", line 286, in __call__
02:21:53     return self._hookexec(self, self.get_hookimpls(), kwargs)
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 93, in _hookexec
02:21:53     return self._inner_hookexec(hook, methods, kwargs)
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 84, in <lambda>
02:21:53     self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/callers.py", line 187, in _multicall
02:21:53     res = hook_impl.function(*args)
02:21:53   File "/builddir/build/BUILDROOT/python-pytest-rerunfailures-9.1.1-3.el9.noarch/usr/lib/python3.9/site-packages/pytest_rerunfailures.py", line 240, in pytest_runtest_protocol
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 126, in runtestprotocol
02:21:53     reports.append(call_and_report(item, "call", log))
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 215, in call_and_report
02:21:53     call = call_runtest_hook(item, when, **kwds)
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 254, in call_runtest_hook
02:21:53     return CallInfo.from_call(
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 311, in from_call
02:21:53     result: Optional[TResult] = func()
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 255, in <lambda>
02:21:53     lambda: ihook(item=item, **kwds), when=when, reraise=reraise
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/hooks.py", line 286, in __call__
02:21:53     return self._hookexec(self, self.get_hookimpls(), kwargs)
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 93, in _hookexec
02:21:53     return self._inner_hookexec(hook, methods, kwargs)
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 84, in <lambda>
02:21:53     self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/callers.py", line 187, in _multicall
02:21:53     res = hook_impl.function(*args)
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/runner.py", line 162, in pytest_runtest_call
02:21:53     item.runtest()
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/python.py", line 1641, in runtest
02:21:53     self.ihook.pytest_pyfunc_call(pyfuncitem=self)
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/hooks.py", line 286, in __call__
02:21:53     return self._hookexec(self, self.get_hookimpls(), kwargs)
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 93, in _hookexec
02:21:53     return self._inner_hookexec(hook, methods, kwargs)
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/manager.py", line 84, in <lambda>
02:21:53     self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
02:21:53   File "/usr/lib/python3.9/site-packages/pluggy/callers.py", line 187, in _multicall
02:21:53     res = hook_impl.function(*args)
02:21:53   File "/usr/lib/python3.9/site-packages/_pytest/python.py", line 183, in pytest_pyfunc_call
02:21:53     result = testfunction(**testargs)
02:21:53   File "/home/jenkins-agent/workspace/nightly_linux-rhel_release/ws/src/ros2/ros2cli/ros2cli/test/test_flake8.py", line 22, in test_flake8
02:21:53     rc, errors = main_with_errors(argv=[])
02:21:53   File "/home/jenkins-agent/workspace/nightly_linux-rhel_release/ws/install/ament_flake8/lib/python3.9/site-packages/ament_flake8/main.py", line 88, in main_with_errors
02:21:53     report = generate_flake8_report(
02:21:53   File "/home/jenkins-agent/workspace/nightly_linux-rhel_release/ws/install/ament_flake8/lib/python3.9/site-packages/ament_flake8/main.py", line 281, in generate_flake8_report
02:21:53     report.report = style.check_files(paths)
02:21:53   File "/usr/lib/python3.9/site-packages/flake8/api/legacy.py", line 109, in check_files
02:21:53     self._application.run_checks(paths)
02:21:53   File "/usr/lib/python3.9/site-packages/flake8/main/application.py", line 271, in run_checks
02:21:53     self.file_checker_manager.run()
02:21:53   File "/usr/lib/python3.9/site-packages/flake8/checker.py", line 309, in run
02:21:53     self.run_parallel()
02:21:53   File "/usr/lib/python3.9/site-packages/flake8/checker.py", line 275, in run_parallel
02:21:53     for ret in pool_map:
02:21:53   File "/usr/lib64/python3.9/multiprocessing/pool.py", line 448, in <genexpr>
02:21:53     return (item for chunk in result for item in chunk)
02:21:53   File "/usr/lib64/python3.9/multiprocessing/pool.py", line 858, in next
02:21:53     self._cond.wait(timeout)
02:21:53   File "/usr/lib64/python3.9/threading.py", line 312, in wait
02:21:53     waiter.acquire()
```